### DISCUSSION TITLE
Two-Factor authentication

### DISCUSSION THREAD

**Discussion by Tsar06** [jlesage/docker-crashplan-pro#459] [General]
I can't seem to connect to the crashplan server using the Username and Server Adress provided by Crashplan. I can't input password or the two-factor code on that screen as the fields do not exist on this login so not sure how I get in the remote crashplan console from this point?! Anyone with the answer?

**Comment by Tsar06** [comment]
FYI I have tried refreshing with most recent package Crashplan v11.2.1 and Docker Image v24.01.1.
The situation evolved as I no more get the server CRASHPLAN_SERVER_ADDRESS prompt - I only get the username prompt with Continue button and then, nothing happens - I get Unable to sign-in. Can't connect to server. Although I can access the console using the normal https://console.us2.crashplan.com/login/#/login - using password and two-factor code.  Can we use this jlesage docker anymore?

**Comment by jlesage** [comment]
To which value did you set the `CRASHPLAN_SERVER_ADDRESS` environment variable ?

**Comment by Tsar06** [comment]
The one crashplan provided to me in theeir welcome email: clients.us5.crashplan.com:4282

**Reply by jlesage** [reply]
Try to only set the address `clients.us5.crashplan.com`, without the port.

**Comment by Tsar06** [comment]
it then prompts for username - then... shows "signin-in..." service message - then...   "unable to sign-in. Can't connect to server." red on top. I am not sure about the debug log as setting the DEBUG to 1 will not display anything I can see in the docker log from the syno. Maybe I should see in some file on the server?

**Comment by Tsar06** [comment]
latest is 11.2.1_23 which I have well installed.
as per 55-crashplan-pro.sh the <authority address="'$CRASHPLAN_SERVER_ADDRESS'" should set the port given in docker parm. so the :4282 as the SMB or empty you set by default is 4287. 
Do I need to open up some given ports on router and firewall? I have done most indicated on crashplan doc.
Sorry my last please: can we browse the log the crashplan code uses to dump the issue it faces during sign-in?
seems to be ${CRASHPLAN_DIR}/electron/crashplan --no-sandbox >> /config/log/ui_output.log 2>> /config/log/ui_error.log

**Reply by Tsar06** [reply]
Problems seem to start here:
[04.18.24 10:16:08.808 INFO  ice STARTING on.CustomizationUniversalService] ClientCustomization:: Unable to get client customization data. The deploymentPolicyToken was not found. StartUp is stopping.
[04.18.24 10:16:08.815 INFO  ice STARTING e42.service.backup.BackupService] BACKUP:: STARTED Backup service in 106.5 μs
[04.18.24 10:16:08.820 INFO  ice STARTING .code42.service.peer.PeerService] STARTED PeerService
[04.18.24 10:16:08.829 INFO  PAC STARTING proxy.ProxyAutoConfigServiceImpl] PROXY:: STARTED proxy PAC service; refreshInterval=every 1 HOURS}.
[04.18.24 10:16:08.831 INFO  -PAC RUNNING proxy.ProxyAutoConfigServiceImpl] PROXY:: Updating proxy on regular interval
[04.18.24 10:16:08.836 INFO  ice STARTING .service.ui.http.KeystoreService] Keystore entry is still valid.
[04.18.24 10:16:08.848 INFO  ice STARTING e42.service.uiinfo.UIInfoUtility] Loaded address=127.0.0.1, port=4243, token, and validateCertificate=true from location=/var/lib/crashplan/.ui_info
[04.18.24 10:16:08.853 INFO  ice STARTING e42.service.uiinfo.UIInfoService] STARTED UIInfoService
[04.18.24 10:16:08.864 INFO  mpl STARTING ce.auth.ReconnectCustomScheduler] RECONNECT_AUTHORITY:: Reconnection scheduled for 24hr 0min from now. Regularly scheduled.
[04.18.24 10:16:08.897 INFO  ice STARTING .service.ui.http.KeystoreService] Wrote certificate to /var/lib/crashplan/service.pem; writable=true, readable=true
[04.18.24 10:16:08.898 INFO  ice STARTING .service.ui.http.KeystoreService] STARTED KeystoreService
[04.18.24 10:16:08.899 INFO  main         p42.service.ClientServiceManager] CSM:: Finished starting universal services in 106.2 ms.
[04.18.24 10:16:08.900 INFO  main         p42.service.ClientServiceManager] CSM:: -------------------------------------------------------------
[04.18.24 10:16:08.901 INFO  main           com.backup42.service.CPService] Not authorized, skip setting up backup layer.
[04.18.24 10:16:08.941 INFO  main         up42.service.peer.PeerController] PeerNetwork has changed, updating peers.
[04.18.24 10:16:29.112 INFO  main         2.service.ui.http.AuthTokenStore] AUTH_TOKEN:: Remove ALL tokens.
[04.18.24 10:16:29.229 INFO  main         .storage.UniversalStorageService] SM:: Saved ServiceModel of 7412 bytes to UDB in 107.1 ms.
[04.18.24 10:16:29.233 INFO  main         de42.service.ui.http.JettyServer] UI:: Starting Jetty...
[04.18.24 10:16:29.235 INFO  main         e42.service.uiinfo.UIInfoUtility] Loaded address=127.0.0.1, port=4243, token, and validateCertificate=true from location=/var/lib/crashplan/.ui_info
[04.18.24 10:16:29.239 INFO  er1WeDftWkr0 up42.service.peer.PeerController] DENIED! Ignore connect, we are not authorized.
[04.18.24 10:16:29.242 INFO  er1WeDftWkr0 ging.direct.DirectNetworkChannel] Channel became inactive. closedBy=THIS_SIDE, reason='Session close', channel=DIRECT
[04.18.24 10:16:30.186 INFO  main         de42.service.ui.http.JettyServer] UI:: Jetty 12.0.2 LISTENING on 127.0.0.1:4243
[04.18.24 10:16:30.189 INFO  main         om.backup42.service.ui.UIService] UI:: Start PING every 30 SECONDS
[04.18.24 10:16:30.191 INFO  EventBus 0   ction.TamperDetectionServiceImpl] TAMPER:: Initiating Tamper Detection using CRL(true) and cpg-agent-code-signatures-11.2.1.23.jar...
[04.18.24 10:16:30.192 INFO  EventBus 0   ction.TamperDetectionServiceImpl] TAMPER:: Using search path: /usr/local/crashplan/lib
[04.18.24 10:16:30.192 INFO  EventBus 0   2.manifest.InstallationValidator] Begin validating software installation
[04.18.24 10:16:30.222 INFO  EventBus 0   42.manifest.ManifestJarValidator] Retrieved certificate chain: length=3
[04.18.24 10:16:30.223 INFO  EventBus 0   42.manifest.ManifestJarValidator] Validated certificate as signed by Crashplan Software
[04.18.24 10:16:30.291 INFO  main         om.backup42.service.AppLogWriter] WRITE app.log in 100ms
[04.18.24 10:16:30.292 INFO  main           com.backup42.service.CPService] *****************************************************************************************
[04.18.24 10:16:30.293 INFO  main           com.backup42.service.CPService]   READY!    Unauthorized    Listening HTTPS port 4243. Startup in 29.94 s
[04.18.24 10:16:30.293 INFO  main           com.backup42.service.CPService] *****************************************************************************************
[04.18.24 10:16:31.319 INFO  re-event-2-2 abre.SabrePendingChannelListener] SABRE::Channel connect failed for guid 4200, cause=java.net.UnknownHostException: clients.us5.crashplan.com
[04.18.24 10:16:31.340 INFO  re-event-2-2 .handler.ChannelLifecycleHandler] SABRE:: Channel became inactive. closedBy=THIS_SIDE, channel=[id: 0xbd63a8c0], sessionState=RemotePeer-[guid=4200, state=C

**Reply by Tsar06** [reply]
error is Couldn't connect to CPC using authority fallbacks, nothing else to do.

**Reply by jlesage** [reply]
This error indicates that server's DNS name can't be resolved:
```
cause=java.net.UnknownHostException: clients.us5.crashplan.com
```
Can the container access the Internet ?

You can try to login to the container `docker exec -ti <container name> sh` and then:

- Check if server's DNS name can be resolved:
```
nslookup clients.us5.crashplan.com
```

- Check if server can be pinged:
```
ping clients.us5.crashplan.com
```

**Comment by Tsar06** [comment]
ping does not work, correct. Do you think crashplan opened the ping for these servers? Can you check if works on yours?

**Reply by jlesage** [reply]
You are right, ping doesn't work.  However, we can usually use `telnet` instead to test the connectivity.

For example, `telnet central.crashplanpro.com 4287` works for me.  But for some reason `telnet clients.us5.crashplan.com 4282` does not...

**Comment by Tsar06** [comment]
nslookup neither works...
How can we set the docker container to access the network? 
I guess i'll look for this!

**Reply by jlesage** [reply]
Usually nothing special is required to allow containers internet access.  But we don't have details about your system, so it's hard to tell why it would not work for you.

**Reply by Tsar06** [reply]
Unclear really, I have a standard conf with synology, nothing fancy I think.
From the container sh I couldn't ping the server name but ip works. So a problem of dns probably. I didn't see an option to inter -dns option in the docker package front-end. Do we need to amend sometthing post generation?

Why is it looking in 172.0.01:4243, I'm not sure..; this is the ui.log

3.18.2024 14:59:26.131 - info: Name:           CrashPlan
3.18.2024 14:59:26.131 - info: Version:        11.2.1
3.18.2024 14:59:26.131 - info: Options:        sandbox=false, _=[], showMenubar=false, showDesktop=true, isDevEnv=false, isTestEnv=false
3.18.2024 14:59:26.132 - info: Platform:       linux (x64)
3.18.2024 14:59:26.132 - info: Service Info:   /var/lib/crashplan/.ui_info
3.18.2024 14:59:26.132 - info: Log File:       /config/.crashplan/log/ui.log
3.18.2024 14:59:26.133 - info: Cache:          /config/xdg/config/CrashPlan
3.18.2024 14:59:26.133 - info: ************************************************************
3.18.2024 14:59:26.425 - info: Retrying communication with service connect ECONNREFUSED 127.0.0.1:4243 attempt 1
3.18.2024 14:59:27.352 - info: Updating system locale to  en_US
3.18.2024 14:59:27.430 - info: Retrying communication with service connect ECONNREFUSED 127.0.0.1:4243 attempt 2
3.18.2024 14:59:28.435 - info: Retrying communication with service connect ECONNREFUSED 127.0.0.1:4243 attempt 3
3.18.2024 14:59:29.448 - info: Retrying communication with service connect ECONNREFUSED 127.0.0.1:4243 attempt 4
3.18.2024 14:59:30.968 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 5
3.18.2024 14:59:32.12 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 6
3.18.2024 14:59:33.44 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 7
3.18.2024 14:59:34.68 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 8
3.18.2024 14:59:35.91 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 9
3.18.2024 14:59:36.115 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 10
3.18.2024 14:59:37.137 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 11
3.18.2024 14:59:38.159 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 12
3.18.2024 14:59:39.180 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 13
3.18.2024 14:59:40.202 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 14
3.18.2024 14:59:41.222 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 15
3.18.2024 14:59:42.243 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 16
3.18.2024 14:59:43.266 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 17
3.18.2024 14:59:44.300 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 18
3.18.2024 14:59:45.322 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 19
3.18.2024 14:59:46.343 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 20
3.18.2024 14:59:47.361 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 21
3.18.2024 14:59:48.384 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 22
3.18.2024 14:59:49.403 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 23
3.18.2024 14:59:50.422 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 24
3.18.2024 14:59:51.441 - info: Retrying communication with service [{"id":"DELAY_LOGIN","params":[]}] attempt 25
3.18.2024 14:59:52.492 - info: Setting locale after receiving loginSetup to AUTOMATIC_LOCALE
3.18.2024 14:59:52.495 - info: Connected to service, show main window
4.18.2024 14:59:54.119 - info JS Console: Init browser console logging
4.18.2024 14:59:54.121 - info main: Starting application initialization
4.18.2024 14:59:54.128 - info main: Initialized ipc listeners to main process
4.18.2024 14:59:54.131 - info main: Initialized push event interface to service
4.18.2024 14:59:54.133 - info main: Initialized https interface to service
4.18.2024 14:59:54.132 - info ServiceInterface: init() with https://127.0.0.1:4243
3.18.2024 14:59:54.168 - info: Closing splash screen
3.18.2024 14:59:54.169 - info: Showing main window
4.18.2024 14:59:54.177 - info main: Performed localization loading, attempting to connect to the service
4.18.2024 14:59:54.377 - info main: Connected to the service
4.18.2024 14:59:54.378 - info main: Attempting to fetch customizations for un-authenticated state
4.18.2024 14:59:54.382 - info main: System locale en_US did not match the locale returned from the service AUTOMATIC_LOCALE, requesting the correct locale
4.18.2024 14:59:54.410 - info main: Attempting to auto-login
4.18.2024 14:59:54.425 - warn JS Console: Attempting to access localized key before initialization.
4.18.2024 14:59:54.417 - info main: Starting main routing
4.18.2024 14:59:54.426 - warn JS Console: Attempting to access localized key before initialization.
4.18.2024 14:59:54.426 - warn JS Console: Attempting to access localized key before initialization.
4.18.2024 14:59:54.572 - info routes.js: onChange: previousPath=/login with params=none
4.18.2024 14:59:54.572 - info routes.js: onChange: nextPath=/login with params=none
4.18.2024 15:1:59.255 - warn RestAdapter: Error making service request to https://127.0.0.1:4243/v1/LoginConfiguration : Gateway Timeout

**Reply by Tsar06** [reply]
Using docker exec -ti jlesage-crashplan-pro sh
I tested ping www.google.com does not work but ping 142.250.178.132 works, so seems like dns is not linked properly from the docker.
Same thing from the host directly works.
I tried ping clients.us5.crashplan.com from docker does not work and ping 64.207.222.177 or 64.207.222.154 doesn't  work either - but does not work as well from the host directly so ping is not supported on these servers as traceroute works.
In summary I need to find why on my synology (standard) the dns is not active from within the docker installed from the package...
Any ideas is welcomed!

**Reply by Tsar06** [reply]
In addition

docker exec -ti jlesage-crashplan-pro cat /etc/resolv.conf returns
nameserver      192.168.0.1
domain  home

which seems correct as 192.168.0.1 is the dns set on my router (freebox)

weird?

**Reply by jlesage** [reply]
Do you have the firewall enabled ? If yes, try to disable it to see if this changes anything.

I would also suggest to check on the Synology forum. I'm sure other people already had similar problem!

**Reply by Tsar06** [reply]
ok the problem is resolved... 
On Synology, we need to use the DNS Server package which I was not using.
Once this is started, because the docker uses the internal DNS resolution, we need then to add the docker network subnet such as 172.17.0.0 with subnet 255.255.0.0 and the problem get resolved after a container restart...

As per the post from Jack Waters
https://community.synology.com/enu/forum/1/post/154904

Thanks to you for help!

**Comment by Tsar06** [comment]
![image](https://github.com/jlesage/docker-crashplan-pro/assets/39737335/115a96ed-fcf6-43b1-88a6-036beef81e23)
